extensions/CubesterYT/TurboHook.js

```diff
@@ -85,9 +138,95 @@
         return JSON.stringify({ username: DATA });
       } else if (MENU == "icon") {
         return JSON.stringify({ avatar_url: DATA });
+      } else if (MENU == "embed") {
+        const embedDATA = JSON.stringify({ embeds: ["!"] });
+        return embedDATA.replace('"!"', DATA);
+      } else if (MENU == "json") {
+        return `json ${DATA}`;
       }
       return "{}";
     }
+    embedParams ({MENU1, MENU2, DATA}) {
+      DATA = Scratch.Cast.toString(DATA);
+      if (MENU1 == "author") {
+        if (MENU2 == "name") {
+          return JSON.stringify({ author: { name: DATA } });
+        } else if (MENU2 == "url") {
+          return JSON.stringify({ author: { url: DATA } });
+        } else if (MENU2 == "icon url") {
+          return JSON.stringify({ author: { icon_url: DATA } });
+        }
+      } else if (MENU1 == "title" && MENU2 == "empty") {
+        return JSON.stringify({ title: DATA });
+      } else if (MENU1 == "url" && MENU2 == "empty") {
+        return JSON.stringify({ url: DATA });
+      } else if (MENU1 == "description" && MENU2 == "empty") {
+        return JSON.stringify({ description: DATA });
+      } else if (MENU1 == "fields") {
+        if (MENU2 == "name") {
+          return JSON.stringify({ fields: [ { name: DATA } ] });
+        } else if (MENU2 == "value") {
+          return JSON.stringify({ fields: [ { value: DATA } ] });
+        } else if (MENU2 == "inline") {
+          return JSON.stringify({ fields: [ { inline: DATA } ] });
+        }
+      } else if (MENU1 == "thumbnail") {
+        if (MENU2 == "url") {
+          return JSON.stringify({ thumbnail: { url: DATA } });
+        }
+      } else if (MENU1 == "image") {
+        if (MENU2 == "url") {
+          return JSON.stringify({ image: { url: DATA } });
+        }
+      } else if (MENU1 == "footer") {
+        if (MENU2 == "text") {
+          return JSON.stringify({ footer: { text: DATA } });
+        } else if (MENU2 == "icon url") {
+          return JSON.stringify({ footer: { icon_url: DATA } });
+        }
+      } else if (MENU1 == "timestamp" && MENU2 == "empty") {
+        return JSON.stringify({ timestamp: DATA });
+      } else if (MENU1 == "color" && MENU2 == "empty") {
+        const color = JSON.stringify(hexToDecimal(DATA));
+        return JSON.stringify({ color: color });
+      } else {
+        return "{}";
+      }
+    }
+    embedConnector ({STRING1, STRING2}) {
+      STRING1 = Scratch.Cast.toString(STRING1);
+      STRING2 = Scratch.Cast.toString(STRING2);
+      if (STRING1.includes("author") && STRING2.includes("author")) {
```

Comment:
making the same mistakes as you did in the original version

you can make this work without duplicating all sorts of code

Reply:
believe what you want is something like "js deep merge objects", but also be careful of "prototype pollution" bugs